ServerCore/Pages/Teams/List.cshtml.cs

```diff
@@ -32,7 +32,7 @@ public async Task<IActionResult> OnGetAsync()
                 return Challenge();
             }
 
-            if (!(await LoggedInUser.IsRegisteredForEvent(_context, Event)))
+            if (Event.HasIndividualRegistration && !(await LoggedInUser.IsRegisteredForEvent(_context, Event)))
```

Comment:
Does a similar check need to happen in Team/Apply.cs?
![image](https://github.com/PuzzleServer/mainpuzzleserver/assets/4121745/e47a3846-08a9-4a2a-889c-1ee80373fa2f)

Right now, it looks like if the player is not registered, they won't be able to apply for a team and will get redirected to register as a player